### PR TITLE
chore:  frame locator repr

### PR DESCRIPTION
**Pull request by kumaraditya303**
Closes https://github.com/microsoft/playwright-python/issues/1005

### CODE REVIEW

**Review by mxschmitt — APPROVED** [review]

File: playwright/_impl/_page.py [hunk 1]
```diff
@@ -199,7 +199,7 @@ def __init__(
         )
 
     def __repr__(self) -> str:
-        return f"<Page url={self.url!r}>"
+        return f"<Page url={self.url!r}> frame={self.main_frame!r}"
```
Comment (mxschmitt) [inline playwright/_impl/_page.py:202]: I think the frame does not provide much value here.

Comment (kumaraditya303) [reply playwright/_impl/_page.py:202]: Frame has name which can be useful.

File: tests/sync/test_sync.py [hunk 2]
```diff
@@ -43,7 +43,10 @@ def test_sync_query_selector(page: Page) -> None:
 
 def test_page_repr(page: Page) -> None:
     page.goto("https://example.com")
-    assert repr(page) == f"<Page url={page.url!r}>"
+    assert (
+        repr(page)
+        == "<Page url='https://example.com/'> frame=<Frame name= url='https://example.com/'>"
```
Comment (mxschmitt) [inline tests/sync/test_sync.py:48]: In my opinion is the page's main frame name always empty, thats why I think we don't have to add the frame to the Page repr.

Comment (kumaraditya303) [reply tests/sync/test_sync.py:48]: If it's always empty then why does that even exists ?

Comment (mxschmitt) [reply tests/sync/test_sync.py:48]: Only sub-frames of a page can have named frames. The main frame's name is always an empty string.

**Review by mxschmitt — APPROVED** [review]

**Pull request merged**